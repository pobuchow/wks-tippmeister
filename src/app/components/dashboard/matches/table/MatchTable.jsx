import React from "react";

export const MatchTable = ({ label, match }) =>
  match ? (
    <div>
      <div className="page-body-label">{label}</div>
      <div className="table-frame">
        <div className="table-div">
          <table className="table">
            <thead>
              <tr>
                <th className="header">Home</th>
                <th className="header">
                  {match.event_datetime.toLocaleDateString(undefined, {
                    year: "numeric",
                    month: "2-digit",
                    day: "2-digit",
                  })}
                </th>
                <th className="header">
                  {match.event_datetime.toLocaleTimeString(undefined, {
                    hour: "2-digit",
                    minute: "2-digit",
                  })}
                </th>
                <th className="header">Away</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td className="match-row">
                  <div className="match-home-team-row">
                    <div className="match-team-icon">
                      <img
                        className="icon-img"
                        src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                        alt=""
                      />
                    </div>
                    <div className="match-home-team-row-name">
                      <p className="match-home-team-row-name-p">
                        {match.homeTeam}
                      </p>
                    </div>
                  </div>
                </td>
                <td className="match-table-result-row">
                  <p className="match-result-p">
                  {match.goalsHomeTeam ? match.goalsHomeTeam : '?'}
                  </p>
                </td>
                <td className="match-table-result-row">
                  <p className="match-result-p">
                  {match.goalsAwayTeam ? match.goalsAwayTeam : '?'}
                  </p>
                </td>
                <td className="match-row">
                  <div className="match-away-team-row">
                    <div className="match-away-team-row-name">
                      <p className="match-away-team-row-name-p">
                        {match.awayTeam}
                      </p>
                    </div>
                    <div className="match-team-icon">
                      <img
                        className=".icon-img"
                        src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                        alt=""
                      />
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  ) : null;