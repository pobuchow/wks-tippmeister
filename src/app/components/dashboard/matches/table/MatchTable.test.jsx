import TestRenderer from 'react-test-renderer';
import React from 'react';
import MatchTable from './MatchTable';

const match = {
  id: '1',
  event_datetime: new Date(2020, 12-1, 17, 18, 0, 0, 0),
  homeTeam: 'Śląsk Wrocław',
  awayTeam: 'Warta Poznań',
  goalsHomeTeam: 2,
  goalsAwayTeam: 1,
};

const matchWithoutGoals = {
  id: '1',
  event_datetime: new Date(2020, 12-1, 17, 18, 0, 0, 0),
  homeTeam: 'Śląsk Wrocław',
  awayTeam: 'Warta Poznań',
  goalsHomeTeam: null,
  goalsAwayTeam: null,
};

const label = 'test_label';

describe('Match Table', () => {
  it('renders snapshot element with goal results', () => {
    const tree = TestRenderer
      .create(<MatchTable label={label} match={match} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not render when match is null', () => {
    const tree = TestRenderer
      .create(<MatchTable label={label} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render snapshot element without goal results', () => {
    const tree = TestRenderer
      .create(<MatchTable label={label} match={matchWithoutGoals} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
