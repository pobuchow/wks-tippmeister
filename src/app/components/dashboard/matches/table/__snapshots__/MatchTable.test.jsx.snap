// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Table does not render when match is null 1`] = `null`;

exports[`Match Table render snapshot element without goal results 1`] = `
<div>
  <div
    className="page-body-label"
  >
    test_label
  </div>
  <div
    className="table-frame"
  >
    <div
      className="table-div"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              className="header"
            >
              Home
            </th>
            <th
              className="header"
            >
              2020-12-17
            </th>
            <th
              className="header"
            >
              18:00
            </th>
            <th
              className="header"
            >
              Away
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="match-row"
            >
              <div
                className="match-home-team-row"
              >
                <div
                  className="match-team-icon"
                >
                  <img
                    alt=""
                    className="icon-img"
                    src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                  />
                </div>
                <div
                  className="match-home-team-row-name"
                >
                  <p
                    className="match-home-team-row-name-p"
                  >
                    Śląsk Wrocław
                  </p>
                </div>
              </div>
            </td>
            <td
              className="match-table-result-row"
            >
              <p
                className="match-result-p"
              >
                ?
              </p>
            </td>
            <td
              className="match-table-result-row"
            >
              <p
                className="match-result-p"
              >
                ?
              </p>
            </td>
            <td
              className="match-row"
            >
              <div
                className="match-away-team-row"
              >
                <div
                  className="match-away-team-row-name"
                >
                  <p
                    className="match-away-team-row-name-p"
                  >
                    Warta Poznań
                  </p>
                </div>
                <div
                  className="match-team-icon"
                >
                  <img
                    alt=""
                    className=".icon-img"
                    src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                  />
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Match Table renders snapshot element with goal results 1`] = `
<div>
  <div
    className="page-body-label"
  >
    test_label
  </div>
  <div
    className="table-frame"
  >
    <div
      className="table-div"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              className="header"
            >
              Home
            </th>
            <th
              className="header"
            >
              2020-12-17
            </th>
            <th
              className="header"
            >
              18:00
            </th>
            <th
              className="header"
            >
              Away
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="match-row"
            >
              <div
                className="match-home-team-row"
              >
                <div
                  className="match-team-icon"
                >
                  <img
                    alt=""
                    className="icon-img"
                    src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                  />
                </div>
                <div
                  className="match-home-team-row-name"
                >
                  <p
                    className="match-home-team-row-name-p"
                  >
                    Śląsk Wrocław
                  </p>
                </div>
              </div>
            </td>
            <td
              className="match-table-result-row"
            >
              <p
                className="match-result-p"
              >
                2
              </p>
            </td>
            <td
              className="match-table-result-row"
            >
              <p
                className="match-result-p"
              >
                1
              </p>
            </td>
            <td
              className="match-row"
            >
              <div
                className="match-away-team-row"
              >
                <div
                  className="match-away-team-row-name"
                >
                  <p
                    className="match-away-team-row-name-p"
                  >
                    Warta Poznań
                  </p>
                </div>
                <div
                  className="match-team-icon"
                >
                  <img
                    alt=""
                    className=".icon-img"
                    src="https://upload.wikimedia.org/wikipedia/en/8/8f/Slask_Wroclaw_crest.png"
                  />
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
